environment/src/main/java/com/sequenceiq/environment/environment/service/FreeipaService.java

```diff
@@ -26,29 +32,64 @@
+        this.freeipaPollerProvider = freeipaPollerProvider;
+        this.webApplicationExceptionMessageExtractor = webApplicationExceptionMessageExtractor;
+    }
+
+    public void startAttachedFreeipaInstances(Long envId, String envCrn) {
+        executeFreeIpaOperationAndStartPolling(envCrn, this::startFreeIpa, freeipaPollerProvider.startPoller(envId, envCrn));
     }
 
-    public void stopAttachedFreeipa(Long envId, String envCrn) {
-        pollingFreeipa(envCrn, freeIpaV1Endpoint::stop, freeipaPollerCollection.stopPoller(envId, envCrn));
+    public void stopAttachedFreeipaInstances(Long envId, String envCrn) {
+        executeFreeIpaOperationAndStartPolling(envCrn, this::stopFreeIpa, freeipaPollerProvider.stopPoller(envId, envCrn));
     }
 
-    public void startAttachedFreeipa(Long envId, String envCrn) {
-        pollingFreeipa(envCrn, freeIpaV1Endpoint::start, freeipaPollerCollection.startPoller(envId, envCrn));
+    private void startFreeIpa(String environmentCrn) {
+        try {
+            freeIpaV1Endpoint.start(environmentCrn);
+        } catch (WebApplicationException e) {
+            String errorMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
+            LOGGER.error("Failed to start FreeIpa(s) for environment '{}' due to: '{}'", environmentCrn, errorMessage, e);
+            throw new EnvironmentServiceException(errorMessage, e);
+        }
+    }
+
+    private void stopFreeIpa(String environmentCrn) {
+        try {
+            freeIpaV1Endpoint.stop(environmentCrn);
+        } catch (WebApplicationException e) {
+            String errorMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
+            LOGGER.error("Failed to stop FreeIpa(s) for environment '{}' due to: '{}'", environmentCrn, errorMessage, e);
+            throw new EnvironmentServiceException(errorMessage, e);
+        }
     }
 
-    private void pollingFreeipa(String envCrn, Consumer<String> fn, AttemptMaker<Void> attemptMaker) {
-        DescribeFreeIpaResponse describeFreeipa = freeipaPollerCollection.describe(envCrn);
-        if (describeFreeipa != null && !describeFreeipa.getStatus().isAvailable()) {
-            fn.accept(envCrn);
+    private void executeFreeIpaOperationAndStartPolling(String envCrn, Consumer<String> freeIpaOperation, AttemptMaker<Void> attemptMaker) {
+        try {
+            DescribeFreeIpaResponse describeFreeipa = freeipaPollerProvider.describe(envCrn);
+            if (describeFreeipa != null && !describeFreeipa.getStatus().isAvailable()) {
+                freeIpaOperation.accept(envCrn);
+                Polling.stopAfterAttempt(attempt)
+                        .stopIfException(true)
+                        .waitPeriodly(sleeptime, TimeUnit.SECONDS)
+                        .run(attemptMaker);
+            }
+        } catch (PollerException e) {
+            if (e.getCause() != null && e.getCause() instanceof WebApplicationException) {
```

Comment:
either rethrow this exception or leave a comment in the code if the condition is false